### PR TITLE
CA-177402 - XenCenter doesnt show migrate option on selecting multipl…

### PR DESCRIPTION
**Pull request by letsboogey**
…e Suspended VMs

Signed-off-by: Letsibogo Ramadi <letsibogo.ramadi@citrix.com>

### CODE REVIEW

File: XenAdmin/Commands/MoveVMCommand.cs [hunk 1]
```diff
@@ -66,16 +66,22 @@ public MoveVMCommand(IMainWindow mainWindow, VM vm)
 
         protected override void ExecuteCore(SelectedItemCollection selection)
         {
-            VM vm = (VM)selection[0].XenObject;
 
-            if (CrossPoolMoveVMCommand.CanExecute(vm, null))
+            if (selection.AllItemsAre<VM>() && new CrossPoolMoveVMCommand(MainWindowCommandInterface, selection).CanExecute())
```
Comment (letsboogey) [inline XenAdmin/Commands/MoveVMCommand.cs]: Yep you are right, it's not necessary there. It makes sense to check if selection contains more than one VM to use that execution path

File: XenAdmin/Commands/MoveVMCommand.cs [hunk 2]
```diff
@@ -67,7 +67,7 @@ public MoveVMCommand(IMainWindow mainWindow, VM vm)
         protected override void ExecuteCore(SelectedItemCollection selection)
         {
 
-            if (selection.AllItemsAre<VM>() && new CrossPoolMoveVMCommand(MainWindowCommandInterface, selection).CanExecute())
+            if (selection.Count > 1 && new CrossPoolMoveVMCommand(MainWindowCommandInterface, selection).CanExecute())
```
Comment (MihaelaStoica) [inline XenAdmin/Commands/MoveVMCommand.cs]: Actually, the count check is not needed, we want the CrossPoolMoveVMCommand even if there is only one VM selected.

Comment (letsboogey) [inline XenAdmin/Commands/MoveVMCommand.cs]: If only one VM is selected don't we want to use the MoveVMDialog instead?

Comment (kc284) [inline XenAdmin/Commands/MoveVMCommand.cs]: If I'm not mistaken on older servers we launch the MoveVMDialog which moves one VM at a time.

Comment (MihaelaStoica) [inline XenAdmin/Commands/MoveVMCommand.cs]: The MoveVMDialog does the intra-pool move of a VM's disks. But if a cross-pool can be done, then we should offer this, even for a single VM.

Comment (kc284) [inline XenAdmin/Commands/MoveVMCommand.cs]: Ok, agreed.

Comment (letsboogey) [inline XenAdmin/Commands/MoveVMCommand.cs]: Ok okay, will make the change

**Pull request merged**